Fix a bug blocking support of 8.2.2 in pytest-rerunfailures:
https://github.com/pytest-dev/pytest-rerunfailures/issues/267

It fails with
> /Python-bundle-PyPI/2024.06-GCCcore-13.3.0/lib/python3.12/site-packages/_pytest/unittest.py", line 319, in runtest
>    assert testcase is not None

From https://github.com/pytest-dev/pytest/pull/12438

From 31791676501f8a8f9eb0d316a6d841a4c0d6738d Mon Sep 17 00:00:00 2001
From: Ran Benita <ran@unusedvar.com>
Date: Sat, 8 Jun 2024 02:11:20 +0300
Subject: [PATCH] [8.2.x] unittest: fix assertion errors on unittest reruns
---
diff --git a/src/_pytest/unittest.py b/src/_pytest/unittest.py
index baca58bd9b2..2f3fd5b014c 100644
--- a/src/_pytest/unittest.py
+++ b/src/_pytest/unittest.py
@@ -216,7 +216,7 @@ def teardown(self) -> None:
             self._explicit_tearDown()
             self._explicit_tearDown = None
         self._obj = None
-        self._instance = None
+        del self._instance
         super().teardown()
 
     def startTest(self, testcase: "unittest.TestCase") -> None:
